internal/registry/revisions.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package registry
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/Masterminds/semver"
+	"github.com/pkg/errors"
+
+	"github.com/google/go-querystring/query"
+
+	"github.com/elastic/elastic-package/internal/packages"
+)
+
+// SearchOptions specify the query parameters for the search API
+type SearchOptions struct {
+	Internal     bool   `url:"internal"`
+	Experimental bool   `url:"experimental"`
+	All          bool   `url:"all"`
+	Package      string `url:"package"`
```

Comment:
I just noticed that `Package` is leaking outside. WDYT about extracting `SearchOptions` (without Package) and adding a new structure `searchQuery` that combines  `SearchOptions` and `Package`.